controllers/cluster_controller_test.go

```diff
@@ -3876,6 +3876,82 @@ var _ = Describe(string(fdbtypes.ProcessClassClusterController), func() {
 			})
 		})
 	})
+
+	Describe("GetPublicIPs", func() {
+		var instance FdbInstance
+
+		BeforeEach(func() {
+			err := internal.NormalizeClusterSpec(&cluster.Spec, internal.DeprecationOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			instance = FdbInstance{}
+		})
+
+		Context("with a default pod", func() {
+			BeforeEach(func() {
+				pod, err := GetPod(cluster, "storage", 1)
+				Expect(err).NotTo(HaveOccurred())
+				pod.Status.PodIP = "1.1.1.1"
+				pod.Status.PodIPs = []corev1.PodIP{
+					{IP: "1.1.1.2"},
+					{IP: "2001:db8::ff00:42:8329"},
+				}
+				instance.Pod = pod
+				instance.Metadata = &pod.ObjectMeta
+			})
+
+			It("should be the public IP from the pod", func() {
+				result := instance.GetPublicIPs()
+				Expect(result).To(Equal([]string{"1.1.1.1"}))
+			})
+		})
+
+		Context("with a pod IP pattern configured", func() {
+			BeforeEach(func() {
+				pattern := ":"
+				cluster.Spec.Routing.PodIPPattern = &pattern
+				pod, err := GetPod(cluster, "storage", 1)
+				Expect(err).NotTo(HaveOccurred())
+				pod.Status.PodIP = "1.1.1.1"
+				pod.Status.PodIPs = []corev1.PodIP{
+					{IP: "1.1.1.2"},
+					{IP: "2001:db8::ff00:42:8329"},
```

Comment:
What happens here if we have an IPv4 address in an IPv6 address? Or any other IPv6 formats that might be confused for the wrong address family here.

Reply:
I've updated it to use the built-in IP parsing.  Does that address this concern?

Reply:
Yes, thanks!